src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder.cs

```diff
@@ -53,7 +53,8 @@ internal static partial class DependentTypeFinder
         private static readonly Func<INamedTypeSymbol, bool> s_isNonSealedClass = t => t?.TypeKind == TypeKind.Class && !t.IsSealed;
         private static readonly Func<INamedTypeSymbol, bool> s_isInterfaceOrNonSealedClass = t => s_isInterface(t) || s_isNonSealedClass(t);
 
-        private static readonly ObjectPool<PooledHashSet<INamedTypeSymbol>> s_setPool1 = PooledHashSet<INamedTypeSymbol>.CreatePool(SymbolEquivalenceComparer.Instance);
+        private static readonly ObjectPool<PooledHashSet<INamedTypeSymbol>> s_symbolSetPool = PooledHashSet<INamedTypeSymbol>.CreatePool(SymbolEquivalenceComparer.Instance);
```

Comment:
❓ why do we store the `INamedTypeSymbol` twice? If we already have a dictionary, do we need the separate `PooledHashSet` as well?

Reply:
they're used for ddifferent purposes.  much of the algorithm just uses the set for temporary storage, keeping track of intermediary values.  The dictoinary is only used for the resultant value though, where we do want htat Project.